messaging/src/main/java/org/axonframework/common/infra/FilesystemStyleComponentDescriptor.java

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.common.infra;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.configuration.Component;
+
+import java.util.*;
+
+/**
+ * A {@link ComponentDescriptor} implementation inspired by filesystem structures. Components are represented as
+ * "directories" with properties as "files" and references as "symbolic links". This creates a clear hierarchical
+ * representation with explicit paths to show component relationships.
+ * <p>
+ * Each component is assigned a path in the virtual filesystem. References to components that have already been
+ * described are represented as symbolic links, indicated with the {@code -> /path/to/component} syntax, preventing
+ * infinite recursion in circular references.
+ * <p>
+ * Example output:
+ * <pre>
+ * /
+ * ├── systemComponent/
+ * │   ├── _ref: 123512
+ * │   ├── _type: EventProcessor
+ * │   ├── name: my-processor
+ * │   └── configuration/
+ * │       ├── _ref: 123513
+ * │       ├── _type: ProcessorConfiguration
+ * │       ├── batchSize: 100
+ * │       └── processor -> /systemComponent
+ * </pre>
+ *
+ * @author Mateusz Nowak
+ * @since 5.0.0
+ */
+public class FilesystemStyleComponentDescriptor implements ComponentDescriptor {
+
+    private static final String ROOT_PATH = "/";
+    private static final String PATH_SEPARATOR = "/";
+
+    private final Map<Object, String> componentPaths;
+    private final Map<String, Object> properties;
```

Comment:
I think that the `properties` value **is** the `describedComponents` or `description`, right? Could we, perchance, rename it to something less generic than `properties`? :angel:

Reply:
Or perhaps call it `processedComponents`, as in the Jackson descriptor!